### PR TITLE
[piecewise graph]: support MiniMax-M2

### PR DESCRIPTION
**Pull request by hzh0425**
## Accuracy Tests

gsm8k

with piecewise
```
100%|██████████| 500/500 [00:13<00:00, 35.98it/s] 
Accuracy: 0.948
Invalid: 0.000
Latency: 13.935 s
Output throughput: 3872.816 token/s
```

without piecewise
```
100%|██████████| 500/500 [00:16<00:00, 30.24it/s] 
Accuracy: 0.946
Invalid: 0.000
Latency: 16.681 s
Output throughput: 3268.322 token/s
```

## Checklist

- [ ] Format your code according to the [Format code with pre-commit](https://docs.sglang.io/developer_guide/contribution_guide.html#format-code-with-pre-commit).
- [ ] Add unit tests according to the [Run and add unit tests](https://docs.sglang.io/developer_guide/contribution_guide.html#run-and-add-unit-tests).
- [ ] Update documentation according to [Write documentations](https://docs.sglang.io/developer_guide/contribution_guide.html#write-documentations).
- [ ] Provide accuracy and speed benchmark results according to [Test the accuracy](https://docs.sglang.io/developer_guide/contribution_guide.html#test-the-accuracy) and [Benchmark the speed](https://docs.sglang.io/developer_guide/contribution_guide.html#benchmark-the-speed).
- [ ] Follow the SGLang code style [guidance](https://docs.sglang.io/developer_guide/contribution_guide.html#code-style-guidance).

## Review Process

1. Ping Merge Oncalls to start the PR flow. See the [PR Merge Process](https://github.com/sgl-project/sglang/blob/main/.github/MAINTAINER.md#pull-request-merge-process).
2. Get approvals from [CODEOWNERS](https://github.com/sgl-project/sglang/blob/main/.github/CODEOWNERS) and other reviewers.
3. Trigger CI tests with [comments](https://docs.sglang.io/developer_guide/contribution_guide.html#how-to-trigger-ci-tests) or contact authorized users to do so.
   - `/tag-run-ci-label`, `/rerun-failed-ci`, `/tag-and-rerun-ci`
4. After green CI and required approvals, ask Merge Oncalls to merge.

### CODE REVIEW

File: python/sglang/srt/models/minimax_m2.py [hunk 1]
```diff
@@ -442,9 +443,14 @@ def op_select_experts(self, state):
         hidden_states = state.hidden_states_mlp_input
 
         if router_logits is not None:
-            with get_global_expert_distribution_recorder().with_current_layer(
-                self.layer_id
-            ):
+            ctx = (
```
Comment (Oasis-Git) [inline python/sglang/srt/models/minimax_m2.py:446]: Maybe we can add a TODO comment so that after pcg enable default we can discuss how to fix the ctx

**Review by Oasis-Git — APPROVED** [review]
LGTM. Let's wait for the CI pass
